p2p/discovery/mdns.go

```diff
@@ -116,31 +116,33 @@ func (m *mdnsService) Close() error {
 
 func (m *mdnsService) pollForEntries(ctx context.Context) {
 
-	ticker := time.NewTicker(m.interval)
-	for {
-		select {
-		case <-ticker.C:
-			entriesCh := make(chan *mdns.ServiceEntry, 16)
-			go func() {
-				for entry := range entriesCh {
-					m.handleEntry(entry)
-				}
-			}()
-
-			log.Debug("starting mdns query")
-			qp := &mdns.QueryParam{
-				Domain:  "local",
-				Entries: entriesCh,
-				Service: m.tag,
-				Timeout: time.Second * 5,
+	for ticker := time.NewTicker(m.interval); ; {
```

Comment:
It would reach much nicer if left the ticker initialization out of the loop and just use `for { ...}`.

So:
```
ticker := ...
for {
  ...
}
```

Reply:
It would also simplify the diff.